PR-CHECKLIST.md

```diff
@@ -0,0 +1,19 @@
+# Checklist for merging PRs
+
+Before merging a PR the PR should follow these rules:
+
+* The PR does not have the label "hold"
+* There is an associated GitHub Issue linked to the PR
+* There is sufficient test coverage
+* Required documentation is included in the PR or an associated PR
+* There are no negative reviews, or unresolved comments
+* The PR does not contain changes not relevant to the GitHub Issue
+* PRs has been reviewed by the relevant team, and approved by either a global maintainer or a team maintainer
+* All required status checks have passed successfully
+* If the PR is affected by unstable workflows or tests verify the issues are not introduced by the PR
+
+Merging a PR:
+
+* In most cases a PR should be merged by a team maintainer in the affected area
+* Global maintainers can merge any PRs, but this is considered a fallback
+* If a PR affects multiple areas it can be merged by any of the affected team maintainers
```

Comment:
Based on this definition, any PR, which change anything under `testsuite` can be merged by `@keycloak/testing-maintainers`. This affects almost any PR affecting "core" as most of the PR adds/updates some tests in the testsuite. This also applies to most of the PR from the community. Not sure if it is expected?

Should it be rather `all of` instead of `any` ? Or at least require reviews from *all* changed areas and afterwards anyone from any area can finally merge?

For example if I send PR, which affects all of "core", "testsuite", "admin console" and "docs", I likely need review from all of "Core-maintainers", "UI maintainers" and "Docs maintainers". Not sure about "testsuite maintainers" here (likely not as writing the test related to some OIDC stuff can be still more related to "core" maintainer rather than "testsuite" maintainer).

Reply:
I am reading in the google document that you indeed meant:
- PR should be reviewed *by all* maintainers of the affected areas
- PR can be merged *by any* (as long as it was reviewed by all applicable teams). Still wonder if to clarify it a bit more in this last point and add something that PR can be merged after it was reviewed by all required owners?

Reply:
Added the line @ssilvert suggested to clarify this part

Reply:
@stianst @ssilvert This makes it clearer, Thanks! I am resolving this.